File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
cnt = 0
def solution(numbers, target):
    global cnt
    solve(numbers, target, 0, 0)
    return cnt

def solve(numbers, target, index, sum):
    global cnt
    if index == len(numbers):
        if sum == target:
            cnt+=1
            return
        else:
            return
        
    solve(numbers, target, index+1, sum + numbers[index])
    solve(numbers, target, index+1, sum - numbers[index])

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    static int answer;
    public int solution(int[] numbers, int target) {
         answer = 0;
        recur(0,numbers,target,0);
        
        return answer;
    }
    static void recur(int depth,int[] numbers,int target, int sum){
        if(depth==numbers.length){
            if(sum==target){
                answer++;
            }
            return;
        }
        recur(depth+1,numbers,target,sum+numbers[depth]);
        recur(depth+1,numbers,target,sum-numbers[depth]);
    }
}